tree/tree/src/TBranch.cxx

```diff
@@ -1530,10 +1529,43 @@ Int_t TBranch::GetBulkEntries(Long64_t entry, TBuffer &user_buf)
    return N;
 }
 
-// TODO: Template this and the call above; only difference is the TLeaf function (ReadBasketFast vs
-// ReadBasketSerialized
+////////////////////////////////////////////////////////////////////////////////
+/// Read as many events as possible into the given buffer, using zero-copy
+/// mechanisms.
+///
+/// Returns -1 in case of a failure.  On success, returns a (non-zero) number of
+/// events of the type held by this branch currently in the buffer.
+///
+/// On success, the caller still need to deserialize the content.  For example for
+/// a scalar branch and `N` the return value (i.e. number of entries)
+///
+///     rawdata = static_cast<char*>(buf.GetCurrent());
+///     for (std::size_t i = 0u; i < N; ++i, ++target)
+///         frombuf(rawdata, target); // `frombuf` also advances the `data` pointer
+///
+/// where T is the type stored on this branch.
+///
+/// When `count_buf` points to a valid TBuffer and the branch has a branch count,
+/// `count_buf` will be filled (via a call to GetEntriesSerialized) with the data
+/// from the branchCount.  After deserialization those value can be used to calculate
+/// the number of elements corresponding to each entries.
+///
+/// For each entry the number of elements is the multiplication of
+///    TLeaf *leaf = dynamic_cast<TLeaf*>(branch->GetListOfLeaves()->At(0))
+///    auto len = leaf->GetLen();
+/// and the value in the BranchCount corresponding to that entry (can be obtained
+/// from branch->GetBranchCount())
+///
+/// NOTES:
+/// - This interface is meant to be used by higher-level, type-safe wrappers, not
+///   by end-users.
+/// - This only returns events
```

Comment:
here and above, what does "This only returns events" imply? as opposed to..?

Reply:
That's a good question.  I am not sure what the original author meant ....